src/week-view.c

```diff
@@ -196,44 +211,91 @@ static void week_view_draw(WeekView* wv, cairo_t* cr)
-		cairo_set_font_size(cr, 22);
-		cairo_set_source_rgb(cr, dark, dark, dark);
-		cairo_show_text(cr, daylabel);
+		// day of month
+		strftime(day_label, 16, "%e", t);
+		cairo_move_to(cr, x + 8, wv->y + HEADER_HEIGHT - 14);
+		cairo_set_font_size(cr, 14);
+
+		if (day.month == wv->now.month && day.day == wv->now.day) {
+			gdk_cairo_set_source_rgba(cr, color_current_day);
+		} else {
+			gdk_cairo_set_source_rgba(cr, color_fg);
+		}
+		cairo_show_text(cr, day_label);
 
-		strftime(daylabel, 16, "%a", t);
+		// weekday abbreviation
+		strftime(day_label, 16, "%a", t);
 
-		for (char* p = daylabel; *p; ++p)
+		for (char* p = day_label; *p; ++p)
 			*p = toupper(*p);
-		cairo_move_to(cr, x + 35, wv->y + HEADER_HEIGHT - 5);
-		cairo_set_font_size(cr, 11);
-		cairo_show_text(cr, daylabel);
+		cairo_move_to(cr, x + 30, wv->y + HEADER_HEIGHT - 14);
+		cairo_set_font_size(cr, 14);
+		cairo_show_text(cr, day_label);
 
-		cairo_set_source_rgb(cr, med, med, med);
+		gdk_cairo_set_source_rgba(cr, color_fg_50);
 		cairo_move_to(cr, x, wv->y + HEADER_HEIGHT);
 		cairo_rel_line_to(cr, 0, wv->height);
 		cairo_stroke(cr);
+
+		gdk_cairo_set_source_rgba(cr, color_header_divider);
+		cairo_move_to(cr, x, wv->y);
+		cairo_rel_line_to(cr, 0, HEADER_HEIGHT);
+		cairo_stroke(cr);
 	}
 
 	// top bar
-	cairo_set_source_rgb(cr, dark, dark, dark);
+	gdk_cairo_set_source_rgba(cr, color_bg);
 	cairo_move_to(cr, wv->x, wv->y + HEADER_HEIGHT);
 	cairo_rel_line_to(cr, wv->width, 0);
 	cairo_move_to(cr, wv->x, wv->y + day_begin_yoffset);
 	cairo_rel_line_to(cr, wv->width, 0);
 	cairo_stroke(cr);
 }
 
+void allocate_colors(void) {
+	color_bg = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_header_bg = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_header_divider = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_fg = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_fg_50 = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_current_time_marker = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+	color_current_day = (GdkRGBA *)malloc(sizeof(GdkRGBA));
+}
+
+void init_style(GtkWidget *mainWindow) {
+	GtkStyleContext* sc = gtk_widget_get_style_context(mainWindow);
+	GdkRGBA color;
+	gtk_style_context_get_color(sc, GTK_STATE_FLAG_NORMAL, &color);
+	gboolean is_dark = color.red > 0.5 && color.blue > 0.5 && color.green > 0.5;
+	if (is_dark) {
+		gdk_rgba_parse(color_bg, "#444444");
+		gdk_rgba_parse(color_header_bg, "#333333");
+		gdk_rgba_parse(color_header_divider, "#666666");
+		gdk_rgba_parse(color_fg, "#aaaaaa");
+		gdk_rgba_parse(color_fg_50, "#808080");
+		gdk_rgba_parse(color_current_time_marker, "#ff8f7e");
+		gdk_rgba_parse(color_current_day, "#79a8cc");
+	} else {
+		gdk_rgba_parse(color_bg, "#fafbfc");
+		gdk_rgba_parse(color_header_bg, "#dadada");
+		gdk_rgba_parse(color_header_divider, "#b6b6b6");
+		gdk_rgba_parse(color_fg, "#4d4d4d");
+		gdk_rgba_parse(color_fg_50, "#a6a6a6");
+		gdk_rgba_parse(color_current_time_marker, "#ff0000");
+		gdk_rgba_parse(color_current_day, "#356797");
```

Comment:
I might have to tweak some of these :)

At some point I think it should be possible to retrieve all these directly from the theme instead of hard-coding them manually, but this is OK for now